### PR TITLE
Remove deprecated constructors for using zeq::URI

### PR DESCRIPTION
**Pull request by bilgili**

### CODE REVIEW

**Comment by tribal-tec** [comment]
Changelog update and version bump missing. +1 otherwise.
Merge should be done once release is out and downstream projects are ready as well.

**Comment by tribal-tec** [comment]
Hold on, not complete yet:

1. zeq::Publisher::getURI() and zeq::http::Server::getURI() must return zeq::URI as well (see todo)
2. zeq::URI( const servus::URI& ) needs to be explicit (see todo)
3. zeq/uri.cpp:28 has TODO to be resolved

**Comment by bilgili** [comment]
So, additionally all "todos" have to be resolved for this topic.

**Comment by tribal-tec** [comment]
The mentioned todos are part of this topic, yes. Not sure if it's all todos in the project.

File: tests/uri.cpp [hunk 1]
```diff
@@ -38,14 +38,6 @@ BOOST_AUTO_TEST_CASE(host_port)
     BOOST_CHECK_EQUAL( hostPort.getScheme(), "tcp" );
 }
 
-BOOST_AUTO_TEST_CASE(custom_schema_not_supported_yet)
```
Comment (tribal-tec) [inline tests/uri.cpp]: that test is still valid imo.

Comment (bilgili) [inline tests/uri.cpp]: // TODO: only for compatibility for hbp schemas, remove after downstreams		
// are fixed
I removed this to do from uri.cpp. What is in mind for this then ?

Comment (tribal-tec) [inline tests/uri.cpp]: The schema is left untouched, so actually only line 46 changes. The testname is then custom_schema. For the zeq::URI this is supported then, but not yet for the publisher/subscriber. Will do later then when needed. inproc could be interesting to have 'local' GUI only for instance.

**Pull request merged**